sui-execution/latest/sui-move-natives/src/tx_context.rs

```diff
@@ -54,3 +59,346 @@ pub fn derive_id(
         smallvec![Value::address(address)],
     ))
 }
+
+#[derive(Clone)]
+pub struct TxContextFreshIdCostParams {
+    pub tx_context_fresh_id_cost_base: InternalGas,
+}
+/***************************************************************************************************
+ * native fun fresh_id
+ * Implementation of the Move native function `fun fresh_id(): address`
+ **************************************************************************************************/
+pub fn fresh_id(
```

Comment:
Wait, what even is this function? Is it `fresh_object_address`? Why isn't that kept in Move?

Reply:
see comment about that above (as an asnwer to @tzakian) we need to compute the fresh id in the new model and that is all in rust.
How could we do that in Move code?